lib/api/apiUtils/bucket/models/ReplicationConfiguration.js

```diff
@@ -101,14 +121,13 @@ class ReplicationConfiguration {
             prefix: rule.Prefix[0],
             enabled: rule.Status[0] === 'Enabled',
         };
+        // ID is an optional property, but create one if not provided or is ''.
+        obj.id = rule.ID && rule.ID[0] !== '' ? rule.ID[0] :
```

Comment:
Is `rule.ID` an array when not undefined? In which case if it's an empty array I assume we will affect an undefined element to `obj.id` which may not be the goal here. (an empty array is truish)